tests/test_all_rules.py

```diff
@@ -165,6 +168,37 @@ def test_tactic_to_technique_correlations(self):
                                          f'technique ID {technique["id"]} does not match the reference URL ID '
                                          f'{technique["reference"]}')
 
+                        # sub-techniques
+                        sub_techniques = technique.get('subtechnique')
+                        if sub_techniques:
+                            for sub_technique in sub_techniques:
+                                expected_sub_technique = attack.technique_lookup[sub_technique['id']]['name']
+                                self.assertEqual(expected_sub_technique, sub_technique['name'],
+                                                 f'ATT&CK sub-technique mapping error for rule: {rule.id} - {rule.name} ->\n'  # noqa: E501
+                                                 f'expected: {expected_sub_technique} for {sub_technique["id"]}\n'
+                                                 f'actual: {sub_technique["name"]}')
+
+                                sub_technique_reference_id = '.'.join(
+                                    sub_technique['reference'].rstrip('/').split('/')[-2:])
+                                self.assertEqual(sub_technique['id'], sub_technique_reference_id,
+                                                 f'ATT&CK sub-technique mapping error for rule: {rule.id} - {rule.name} ->\n'  # noqa: E501
+                                                 f'sub-technique ID {sub_technique["id"]} does not match the reference URL ID '  # noqa: E501
+                                                 f'{sub_technique["reference"]}')
+
+    def test_duplicated_tactics(self):
```

Comment:
While I was testing, I noticed there were 8 rules which defined the same tactic multiple times with different techniques, so I added a test to ensure this is no longer done. This is the reason for the other 8 rule changes as well